napari/utils/key_bindings.py

```diff
@@ -182,6 +186,284 @@ def normalize_key_combo(key_combo):
 UNDEFINED = object()
 
 
+def call_with_context(function, context):
+    """
+    call function `function` with the corresponding value taken from context
+
+    This is use in the action manager to pass the rights instances to the actions,
+    without having the need for them to take a **kwarg, and is mostly needed when
+    triggering actions via buttons, or to record.
+
+    If we went a declarative way to trigger action we cannot refer to instances
+    or objects that must be passed to the action, or at least this is
+    problematic.
+
+    We circumvent this by having a context (dictionary of str:instance) in
+    the action manager, and anything can tell the action manager "this is the
+    current instance a key". When an action is triggered; we inspect the
+    signature look at which instances it may need and pass this as parameters.
+    """
+    from inspect import signature
+
+    context_keys = [
+        k
+        for k, v in signature(function).parameters.items()
+        if v.kind not in (v.VAR_POSITIONAL, v.VAR_KEYWORD)
+    ]
+    ctx = {k: v for k, v in context.items() if k in context_keys}
+    return function(**ctx)
```

Comment:
Could we catch any possible errors here and provide a localizable and friendlier message?

Reply:
Maybe, I don't know if we want to do that; or ask code that register actions to be explicit as to which parameters need to be passed in.